File: state/ProcessingOrderState.java
package state;

public class ProcessingOrderState implements OrderState {
    private final OrderManagement orderManagement;

    public ProcessingOrderState(OrderManagement orderManagement) {
        this.orderManagement = orderManagement;
    }

    @Override
    public void processOrder() {
        System.out.println("The order is currently being processed!");
    }

    @Override
    public void shipOrder() {
        System.out.println("Shipping the order...");
        orderManagement.changeState(new ShippedOrderState(orderManagement));
    }

    @Override
    public void deliverOrder() {
        System.out.println("The order cannot be delivered at the current state");
    }

    @Override
    public void cancelOrder() {
        System.out.println("The order is being cancelled now...");
        orderManagement.changeState(new CancelledOrderState(orderManagement));
    }
}

File: PHP/Patterns/src/Behavioral/State/ProcessingOrderState.php
<?php

namespace src\Behavioral\State;

class ProcessingOrderState implements OrderState
{
    private OrderManagement $orderManagement;

    /**
     * @param OrderManagement $orderManagement
     */
    public function __construct(OrderManagement $orderManagement)
    {
        $this->orderManagement = $orderManagement;
    }

    function processOrder()
    {
        print_r("This order is already processed... \n");
    }

    function shipOrder()
    {
        print_r('Order: ' . $this->orderManagement->getOrder()->getName() . " is being shipping now... \n");
        $this->orderManagement->changeState(new ShippedOrderState($this->orderManagement));
    }

    function deliverOrder()
    {
        print_r("The order can't be delivered at the current state... \n");
    }

    function cancelOrder()
    {
        print_r('Order: ' . $this->orderManagement->getOrder()->getName() . " is being cancelling now... \n");
        $this->orderManagement->changeState(new CancelledOrderState($this->orderManagement));
    }
}